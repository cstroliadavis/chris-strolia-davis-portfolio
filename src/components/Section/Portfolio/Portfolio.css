.section.portfolio .app-list {
    column-gap: 1rem;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    margin: 2rem;
}

@media (max-width: 1000px) {
    .section.portfolio .app-list {
        grid-template-columns: 1fr 1fr;
    }
}

@media (max-width: 500px) {
    .section.portfolio .app-list {
        grid-template-columns: 1fr;
    }
}
