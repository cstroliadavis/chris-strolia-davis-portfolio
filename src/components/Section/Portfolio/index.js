import loading from '../../../assets/img/loading-gif.gif';
import './Portfolio.css';
import { useEffect, useState } from "react";
import AppCard from '../../AppCard';
import { getAllApplications } from '../../../services/applications.repository';

export default function Portfolio() {
  const [ isLoading, setIsLoading ] = useState(false);
  const [ applications, setApplications ] = useState([]);
  const [ errors, setErrors ] = useState(null);

  useEffect(() => {
    setIsLoading(true);

    getAllApplications()
      .then((appData) => setApplications(appData))
      .catch(setErrors)
      .finally(() => setIsLoading(false))

    return () => {
    };
  }, []);

  return (
    <div className="container">
      {
        isLoading
          ? <img alt="Loading" src={ loading }/>
          :errors
            ? <div className="error">There was an issue loading data. Please try again</div>
            :<div className="app-list">
              { applications.map(app => (<AppCard { ...app } key={ app.id }/>)) }
            </div>
      }
    </div>
  );
}
