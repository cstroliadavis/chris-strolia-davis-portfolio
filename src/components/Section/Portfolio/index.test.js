import { render, cleanup, act } from '@testing-library/react';
import Portfolio from './index'

beforeAll(jest.useFakeTimers);
afterAll(jest.useRealTimers);

afterEach(cleanup);

const mockApplications = Object.freeze([
  Object.freeze({
    id: 1,
    title: 'Stuff',
    description: 'Things',
    url: 'https://app-url.com',
    image: 'https://image-url.com',
    github: 'https://githb-url.com',
  }),
  Object.freeze({
    id: 2,
    title: 'Mo Stuff',
    description: 'Mo Things',
    url: 'https://app-url.com/mo',
    image: 'https://image-url.com/mo-stuff.png',
    github: 'https://githb-url.com/mo',
  }),
]);

jest.mock('../../../services/applications.repository', () => ({
  getAllApplications: () => new Promise(resolve =>
    setTimeout(() => resolve(mockApplications), 1000)
  )
}));

describe('Portfolio', () => {
  it('renders', async () => {
    act(() => {
      render(<Portfolio/>)
    });
  });

  it('matches snapshot', async () => {
    let res = null;
    act(() => {
      res = render(<Portfolio/>)
    });
    await act(async () => {
      await jest.advanceTimersByTime(1500)
    });
    expect(res.asFragment()).toMatchSnapshot();
  });

  it('shows a spinner while it waits', () => {
    let res = null;
    act(() => {
      res = render(<Portfolio/>)
    });
    expect(res.asFragment()).toMatchSnapshot();
  });
});
