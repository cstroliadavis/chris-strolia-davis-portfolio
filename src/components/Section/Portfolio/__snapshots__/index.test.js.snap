// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="app-list"
    >
      <div
        class="app-card"
      >
        <h1
          class="title"
        >
          Stuff
        </h1>
        <div
          class="app-ref"
        >
          <a
            href="https://app-url.com"
            target="_blank"
          >
            <img
              alt="Stuff"
              src="https://image-url.com"
            />
          </a>
        </div>
        <div
          class="repo"
        >
          <a
            href="https://githb-url.com"
            target="_blank"
          >
            See the code
          </a>
        </div>
        <div
          class="app-description"
        >
          <a
            href="https://app-url.com"
            target="_blank"
          >
            Things
          </a>
        </div>
      </div>
      <div
        class="app-card"
      >
        <h1
          class="title"
        >
          Mo Stuff
        </h1>
        <div
          class="app-ref"
        >
          <a
            href="https://app-url.com/mo"
            target="_blank"
          >
            <img
              alt="Mo Stuff"
              src="https://image-url.com/mo-stuff.png"
            />
          </a>
        </div>
        <div
          class="repo"
        >
          <a
            href="https://githb-url.com/mo"
            target="_blank"
          >
            See the code
          </a>
        </div>
        <div
          class="app-description"
        >
          <a
            href="https://app-url.com/mo"
            target="_blank"
          >
            Mo Things
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Portfolio shows a spinner while it waits 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt="Loading"
      src="loading-gif.gif"
    />
  </div>
</DocumentFragment>
`;
